action/action-simulator/src/main/java/com/powsybl/action/simulator/loadflow/LoadFlowActionSimulator.java

```diff
@@ -107,19 +108,42 @@ public void start(ActionDb actionDb, List<String> contingencyIds) {
 
         observers.forEach(LoadFlowActionSimulatorObserver::afterPreContingencyAnalysis);
 
-        // duplicate the network for each contingency
-        byte[] networkXmlGz = NetworkXml.gzip(network);
+        byte[] networkXmlGz = null;
+        if (!config.isCloneNetworkState()) {
+            // duplicate the network for each contingency
+            networkXmlGz = NetworkXml.gzip(network);
+        }
 
         if (preContingencyAnalysisOk || config.isIgnorePreContingencyViolations()) {
+            String stateId = null;
+            String tmpStateId = null;
             for (String contingencyId : contingencyIds) {
+                if (config.isCloneNetworkState()) {
+                    if (Objects.isNull(tmpStateId)) {
+                        stateId = network.getStateManager().getWorkingStateId();
+                    } else {
+                        network.getStateManager().removeState(tmpStateId);
+                        tmpStateId = null;
+                    }
+                }
+
                 Contingency contingency = actionDb.getContingency(contingencyId);
-                Network network2 = NetworkXml.gunzip(networkXmlGz);
-                RunningContext runningContext2 = new RunningContext(network2, contingency);
+
+                Network network2 = null;
+                if (!config.isCloneNetworkState()) {
+                    network2 = NetworkXml.gunzip(networkXmlGz);
+                } else {
+                    tmpStateId = contingencyId + "-" + UUID.randomUUID();
+                    network.getStateManager().cloneState(stateId, tmpStateId);
+                    network.getStateManager().setWorkingState(tmpStateId);
```

Comment:
i think just pointer network2 to network in this block, then you don't have to check nonNull. 
Netwok2 is always the one to run AS on, just generated by xml or state.
Same for networkForTest